### PR TITLE
Fix trade event watching

### PR DESCRIPTION
**Pull request by Velenir**
- Events had an incorrect topic, switched to using Interface constructed from ABI directly
- Adds filling in of Order.summary for the popup (when Order is in store). TODO: move to Popup middleware
- Fixes non-unique popup.key
localhost_3000_

How to test:
0. Start [here](https://pr82--gpv2swap.review.gnosisdev.com/#/swap?inputCurrency=0xc778417E063141139Fce010982780140Aa0cD5Ab&outputCurrency=0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa)
1. on Rinkeby
2. Sell 0.1 WETH for whatever DAI
3. Sell 90 DAI for WETH
4. Wait

steps courtesy of @fleupold

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr82--gpv2swap.review.gnosisdev.com)**: Testing web app

File: src/custom/state/orders/hooks.ts [hunk 1]
```diff
@@ -57,6 +59,31 @@ export const useOrder = ({ id, chainId }: GetRemoveOrderParams): Order | undefin
   })
 }
 
+// used to extract Order.summary before showing popup
+// TODO: put the whole logic inside Popup middleware
+export const useFindOrderById = ({ chainId }: GetOrdersParams): GetOrderByIdCallback => {
+  const state = useSelector<AppState, OrdersState[ChainId] | undefined>(state => chainId && state.orders?.[chainId])
+
+  // stable ref, so we don't recreate the function
+  const stateRef = useRef(state)
```
Comment (W3stside) [inline src/custom/state/orders/hooks.ts:68]: im lost here, i thought useSelector did this for us

Comment (W3stside) [reply src/custom/state/orders/hooks.ts:68]: oh nvm it's for the useCallback I see.

File: src/custom/state/orders/updater.ts [hunk 2]
```diff
@@ -3,46 +3,32 @@ import { useDispatch, batch } from 'react-redux'
...
 import { buildBlock2DateMap } from 'utils/blocks'
 import { registerOnWindow } from 'utils/misc'
-// import { PartialOrdersMap } from './reducer'
+import { GP_SETTLEMENT_CONTRACT_ADDRESS } from 'constants/index'
+import { GP_V2_SETTLEMENT_INTERFACE } from '@src/custom/constants/GPv2Settlement'
 
-// example of event watching + decoding without contract
-const transferEventAbi = 'event Transfer(address indexed from, address indexed to, uint amount)'
-const ERC20Interface = new utils.Interface([transferEventAbi])
+type OrderLogPupupMixData = OrderFulfillmentData & Pick<Log, 'transactionHash'> & Partial<Pick<Order, 'summary'>>
```
Comment (W3stside) [inline src/custom/state/orders/updater.ts:15]: typo here

File: src/custom/state/orders/updater.ts [hunk 3]
```diff
@@ -144,28 +134,31 @@ export function EventUpdater(): null {
...
         toBlock: lastBlockNumber,
-        // address: '0x', // TODO: get address from networks.json or somewhere else
+        address: contractAddress,
         topics: eventTopics
       })
 
       const block2DateMap = await buildBlock2DateMap(library, logs)
 
-      const ordersBatchData: (OrderFulfillmentData & Pick<Log, 'transactionHash'>)[] = logs.map(log => {
+      const ordersBatchData: OrderLogPupupMixData[] = logs.map(log => {
```
Comment (W3stside) [inline src/custom/state/orders/updater.ts:150]: typo in type name

File: src/custom/state/orders/updater.ts (hunk 2 above)
Comment (W3stside) [reply src/custom/state/orders/updater.ts:15]:
```suggestion
type OrderLogPopupMixData = OrderFulfillmentData & Pick<Log, 'transactionHash'> & Partial<Pick<Order, 'summary'>>
```

File: src/custom/state/orders/updater.ts (hunk 3 above)
Comment (W3stside) [reply src/custom/state/orders/updater.ts:150]:
```suggestion
      const ordersBatchData: OrderLogPopupMixData[] = logs.map(log => {
```

File: src/custom/state/orders/hooks.ts (hunk 1 above)
Comment (Velenir) [reply src/custom/state/orders/hooks.ts:68]: It does return an order, there's even `useOrder` hook
But I want to lookup multiple orders lazily, after we fetch events
If we `useSelector|useOrder`, the order it returns will need to be a dependency of the `useEffect` where fetching takes place, so we'll need to manage state changing there, while we juggle events from a previous fetch

With use Ref we always have a ref to current orders and can lazily search through it

BUT I'll remove the whole thing when I make Popup middleware

**Review by W3stside — APPROVED** [review]

**Review by anxolin — COMMENTED** [review]
👌

**Pull request merged**